### ISSUE TITLE
Memory problems with Geant4 run

### ISSUE THREAD

**Issue by sawenzel** [AliceO2Group/AliceO2#640]
Running `tpc-run-sim -n 1 -e TGeant4` leaves a strange warning message at process exit

```
WARNING - Attempt to delete the physical volume store while geometry closed !
WARNING - Attempt to delete the logical volume store while geometry closed !
WARNING - Attempt to delete the solid store while geometry closed !
WARNING - Attempt to delete the region store while geometry closed !
```

Valgrind also reports several problems with invalid free/delete/ etc. for this run (see attachment [valgrindGeant4.txt](https://github.com/AliceO2Group/AliceO2/files/1417641/valgrindGeant4.txt)).

The same run with Geant3 does not show these problems.

Environment:
FairRoot: af83f7321db
O2: 8b61da9579a287
alidist: 782700a6f17f907

**Comment by sawenzel** [comment]
@ihrivnac : Maybe you have an idea?

**Comment by ihrivnac** [comment]
This happens when you exit your program without deleting VMC application; it can be reproduce with VMC examples macro, eg. E03/run_g4.C when you comment out the last line:
  // delete appl;

**Comment by sawenzel** [comment]
Thanks. I think the problem goes beyond our O2 code and is potentially a problem in FairRoot which manages the VMC application. From an O2 perspective, we do not even see the vmc instance.

**Comment by ihrivnac** [comment]
In FairRoot, the VMC application is created/deleted by FairRunSim, so you should check in O2 whether you delete this object before exiting program.

**Comment by sawenzel** [comment]
yes i do

**Comment by ihrivnac** [comment]
Ok, then it seems to be due to not deleting TVirtualMC object (TGeant4 in this case) in FairRootMCApplication. Could you try to add this delete as the last line in FairMCApplication::~FairMCApplication():
  delete fMC;

**Comment by sawenzel** [comment]
@ihrivnac : Yes this seems to solve part of the problem indeed. I now no longer see the warning but valgrind still reports memory problems.
